### ISSUE TITLE
fread: Error when filename contains middle dot

### ISSUE THREAD

**Issue by ismirsehregal** [Rdatatable/data.table#4801]
There seems to be a problem when trying to read in files having a middle dot in their name via `fread`.
I came across this by trying to read in csv files created by Microsoft Dynamics NAV.
I wasn't able to reproduce this behaviour when the file is written by R (e.g. `fwrite`)

[foo ∙ bar.txt](https://github.com/Rdatatable/data.table/files/5500318/foo.bar.txt) (Unfortunately, this file is renamed to "foo.bar" after downloading from GitHub - please rename it manually)

```
library(data.table)

filename <- list.files(".", "*.txt$")
fread(file = filename)
```

> Error in fread(file = filename) : 
>   File 'foo · bar.txt' does not exist or is non-readable.

Using R version 4.0.3 and data.table 1.13.2 on a Windows 7 machine.

Same behaviour with `utils::read.table`:

> Error in file(file, "rt") : cannot open the connection
> In addition: Warning message:
> In file(file, "rt") :
>   cannot open file 'foo · bar.txt': No such file or directory

**Comment by jangorecki** [comment]
As you are getting the same issue with `read.table` I suggest to report that upstream, in Microsoft Dynamics NAV support. Of course we can try to handle that.

**Comment by ismirsehregal** [comment]
Of course, reporting upstream is a reasonable way to go. I could also write a script to rename those files. I just thought you might be interested regarding the robustness of `fread`.

By now I found out, that I can use `read.table` after capturing the filename via `filename <- file.choose()`. However, this isn't  working with `fread`.

**Comment by ismirsehregal** [comment]
I just tested this with the latest cran version of {data.table}. It seems the issue no longer exists.

Cheers

**Comment by aitap** [comment]
Most likely fixed by [switching R to UTF-8 on Windows](https://blog.r-project.org/2020/05/02/utf-8-support-on-windows/index.html). `data.table` uses the "native" encoding to store file names, which on Windows used to be ANSI, which cannot represent ∙ (U+2219 BULLET OPERATOR), causing `enc2native()` to return the garbled name of the file, thus preventing `fread()` from working. R internals have a special case for this on Windows, losslessly converting file names to Windows-native `wchar_t[]` and using special Windows-only interfaces, which is why `read.table` used to work even before the switch to UTF-8.
